pkg/docker/config/config.go

```diff
@@ -697,7 +696,15 @@ func authKeysForRef(ref reference.Named) (res []string) {
 		name = name[:lastSlash]
 	}
 
-	return res
+	// Named references always contain at least one "/", so in theory we never
+	// return nil in this function. We still preserve this sanity check to
+	// avoid accessing out of bounds in case if future reference
```

Comment:
nit of a nit if you've other changes.
```suggestion
	// avoid accessing out of bounds in case of future reference
```